src/nix/store-check-trusted.cc

```diff
@@ -0,0 +1,61 @@
+#include <sstream>
+
+#include "command.hh"
+#include "logging.hh"
+#include "serve-protocol.hh"
+#include "shared.hh"
+#include "store-api.hh"
+#include "local-fs-store.hh"
+#include "util.hh"
+#include "worker-protocol.hh"
+
+using namespace nix;
+
+namespace {
+
+bool checkPass(const std::string & msg) {
+    if (isatty(fileno(stdin)))
+      notice(ANSI_GREEN "[PASS] " ANSI_NORMAL + msg);
+    return true;
+}
+
+bool checkFail(const std::string & msg) {
+    if (isatty(fileno(stdin)))
+      notice(ANSI_RED "[FAIL] " ANSI_NORMAL + msg);
+    return false;
+}
+
+}
+
+struct CmdCheckTrusted : StoreCommand
+{
+    bool success = true;
+
+    Category category() override { return catUtility; }
+
+    std::string description() override
+    {
+        return "ask a Nix store if you are a trusted-user, return exit code 0 if trusted, 1 if not trusted, 2 if unexpected error";
+    }
+
+    void run(ref<Store> store) override
+    {
+        logger->log("Running checks against store uri: " + store->getUri());
+        success &= checkTrustedUser(store);
+        if (success)
+            throw Exit(0);
+        if (!success)
+            throw Exit(1);
+        throw Exit(2);
+    }
+    bool checkTrustedUser(ref<Store> store)
+    {
+        bool trusted = store->isTrustedUser();
+        if (!trusted) {
+            return checkFail("You are not a trusted-user on store uri: " + store->getUri());
+        }
+        return checkPass("You are a trusted-user on store uri: " + store->getUri());
+    }
+};
+
+static auto rCmdCheckTrusted = registerCommand2<CmdCheckTrusted>({"store", "check-trusted"});
```

Comment:
```suggestion
static auto rCmdCheckTrusted = registerCommand2<CmdCheckTrusted>({"store", "check-trusted-user"});
```

`trusted` has a bunch of different meanings in Nix, better be explicit

Reply:
What other meanings does `trusted` have in Nix? And is this not disambiguated already by the fact that it is a subcommand of `store`?

Reply:
The three that come to my mind are:

- A path is trusted if it has a valid signature or has been built locally
- A public key is trusted or not
- A substituter can also be marked as [trusted](https://nixos.org/manual/nix/stable/command-ref/conf-file.html#conf-trusted-substituters) 

None of these would _totally_ make sense here (except maybe the third one), but they are closely related to the store and I can imagine them being misleading for a user

Reply:
That makes sense, I'll change it.